tests/testthat/test-save-separate-chunks.R

```diff
@@ -0,0 +1,231 @@
+  expect_true(all(c("x", "y", "group") %in% names(common.data)))
+  # randomly choose an varied.chunk to test
+  idx <- sample(no.chunks, 1)
+  varied.data <- read.csv(varied.chunks[idx], sep = "\t")
+  expect_equal(nrow(varied.data), length(unique(USpolygons$group)))
+  expect_true(all(c("fill", "group") %in% names(varied.data)))
+  
+  unlink(out.dir, recursive = TRUE)
+})
+
+
+### test case 2
+USdots <- ddply(USpolygons, .(region), summarise, mean.lat = mean(lat), 
+                mean.long = mean(long))
+# add state flu
+map_flu <- ldply(unique(state_flu$WEEKEND), function(we) {
+  df <- subset(state_flu, WEEKEND == we)
+  merge(USdots, df, by.x = "region", by.y = "state")
+})
+
+test_that("save separate chunks for geom_point without group", {
+  # the compiler will not break a geom into chunks if any of the resulting 
+  # chunk tsv files is estimated to be less than 4KB.
+  state.map <- p + 
+    geom_point(data = map_flu, aes(x = mean.long, y = mean.lat, fill = level, 
+                                   showSelected = WEEKEND), color = "black", size = 10)
+  viz <- list(levelHeatmap = level.heatmap, stateMap = state.map, title = "FluView")
+  out.dir <- file.path(getwd(), "FluView")
+  animint2dir(viz, out.dir = out.dir, open.browser = FALSE)
+  
+  common.chunk <- list.files(path = out.dir, pattern = "geom.+point.+chunk_common.tsv", 
+                             full.names = TRUE)
+  varied.chunks <- list.files(path = out.dir, pattern = "geom.+point.+chunk[0-9]+.tsv", 
+                              full.names = TRUE)
+  # number of chunks
+  expect_equal(length(common.chunk), 0L)
+  expect_equal(length(varied.chunks), 1L)
+  # test the only one varied.chunk
+  varied.data <- read.csv(varied.chunks, sep = "\t")
+  expect_equal(nrow(varied.data), nrow(map_flu))
+  expect_true(all(c("fill", "x", "y", "showSelected") %in% names(varied.data)))
+  
+  unlink(out.dir, recursive = TRUE)
+  
+  ## force to split into chunks
+  state.map <- p + 
+    geom_point(data = map_flu, aes(x = mean.long, y = mean.lat, fill = level, 
+                                   showSelected = WEEKEND), 
+               color = "black", size = 10, chunk_vars = "WEEKEND")
+  viz <- list(levelHeatmap = level.heatmap, stateMap = state.map, title = "FluView")
+  animint2dir(viz, out.dir = out.dir, open.browser = FALSE)
+  
+  common.chunk <- list.files(path = out.dir, pattern = "geom.+point.+chunk_common.tsv", 
+                             full.names = TRUE)
+  varied.chunks <- list.files(path = out.dir, pattern = "geom.+point.+chunk[0-9]+.tsv", 
+                              full.names = TRUE)
+  # number of chunks
+  expect_equal(length(common.chunk), 1L)
+  no.chunks <- length(varied.chunks)
+  expect_equal(no.chunks, length(unique(map_flu$WEEKEND)))
+  # test common.chunk
+  common.data <- read.csv(common.chunk, sep = "\t")
+  expect_equal(nrow(common.data), nrow(USdots))
+  expect_true(all(c("x", "y") %in% names(common.data)))
```

Comment:
you don't expect "group" to be a column name in the geom_point tsv files?

in that case to combine the comomn tsv file with the chunk tsv files, you expect that there are exactly the same number of rows and they are in the same order?